topos/features/v1/features.proto

```diff
@@ -0,0 +1,283 @@
+      returns (TimeSeriesFeatureValuesResponse) {
+    option (google.api.http) = {
+      post : "/v1/feature_values:time_series"
+      body : "*"
+    };
+  }
+}
+
+message FeatureSet {
+  string name = 1;
+  string display_name = 2;
+  string description = 3;
+  bool include_features = 4;
+  repeated Feature features = 5; // features in feature set
+}
+
+message GetFeatureSetRequest {
+  string name = 1;
+
+  // If set true, response includes features of each feature set
+  bool include_features = 2;
+}
+
+message SetFeatureSetRequest { FeatureSet feature_set = 1; }
+
+message DeleteFeatureSetRequest { string name = 1; }
+
+message FeatureSetFeature {
+  string feature_set_name = 1;
+  string feature_name = 2;
+}
+
+message ListFeatureSetsRequest {
+  // The maximum number of items to return.
+  int32 page_size = 1;
+
+  // The next_page_token value returned from a previous Search request, if any.
+  string page_token = 2;
+}
+
+message ListFeatureSetsResponse {
+  string next_page_token = 1;
+  repeated FeatureSet feature_sets = 2;
+}
+
+message Feature {
+  string name = 1;
+  string display_name = 2;
+  string description = 3;
+  string source = 4;
+}
+
+message SetFeatureRequest { Feature feature = 1; }
+
+message DeleteFeatureRequest { string name = 1; }
+
+message ListFeaturesRequest {
+  // The maximum number of items to return.
+  int32 page_size = 1;
+
+  // The next_page_token value returned from a previous Search request, if any.
+  string page_token = 2;
+}
+
+message ListFeaturesResponse {
+  repeated Feature feature = 1;
+  string next_page_token = 2;
+}
+
+enum RegionType {
+  REGION_TYPE_UNSPECIFIED = 0;
+  US_ZIPCODE = 1;
+}
+
+enum TimeType {
+  TIME_TYPE_UNSPECIFIED = 0;
+  EPOCH = 1;
+  YEAR = 2;
+  MONTH = 3;
+  DAY = 4;
+  HOUR = 5;
+  MINUTE = 6;
+}
+
+message FeatureValue {
+  string name = 1;
+  RegionType region_type = 2;
+  string region = 3;
+  TimeType time_type = 4;
+  string time = 5;
+  double value = 6;
+}
+
+message SetFeatureValueRequest { FeatureValue feature_value = 1; }
+
+message BatchSetFeatureValueRequest {
+  repeated FeatureValue feature_values = 1;
+}
+
+message BatchSetFeatureValueResponse {
+  repeated google.rpc.Status status = 1;
+}
+
+message DeleteFeatureValueRequest { string name = 1; }
+
+message FeatureData {
+  string name = 1;
+  double value = 2;
+}
+
+message SearchFeatureValuesRequest {
+  // The maximum number of items to return.
+  int32 page_size = 1;
+
+  // The next_page_token value returned from a previous Search request, if any.
+  string page_token = 2;
+
+  string name = 3;
+  RegionType region_type = 4;
+  string region = 5;
+  TimeType time_type = 6;
+  string time = 7;
+}
+
+message SearchFeatureValuesResponse {
+
+  // Token to retrieve the next page of results, or empty if there are no more
+  // results in the list.
+  string next_page_token = 1;
+
+  string name = 2;
+  RegionType region_type = 3;
+  string region = 4;
+  TimeType time_type = 5;
+  string time = 6;
```

Comment:
The keys here represent data that was sent with the request and can be omitted. See bellow.